lib/plugins/authldap/auth.php

```diff
@@ -550,8 +550,12 @@ protected function openLDAP()
         $bound = false;
         $servers = explode(',', $this->getConf('server'));
         foreach ($servers as $server) {
-            $server = trim($server);
-            $this->con = @ldap_connect($server, $port);
+            if($this->getConf('uri')) {
```

Comment:
getConfig(‘server’) could have multiple servers, probably the same for ‘uri’? Seems not consistent imho.
Is this part only reached if anything in ‘server’ is set? (But further not used)

Reply:
oh, good catch, that really should be outside of the loop, I'm trying to ship off a bunch of older code that I've accumulated, will clean that up.

Reply:
Thinking about this, might be better to just do a string comparison against $server to see if it starts with ldap:// or ldaps://  and then if so, use the single value instead of server/port.

Reply:
Adjusted code to just look for a uri instead, cleaner, and should also now support 'multiple uri's in server attribute'.